src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -3201,9 +3201,11 @@ public override SyntaxNode AwaitExpression(SyntaxNode expression)
 
         public override SyntaxNode NameOfExpression(SyntaxNode expression)
         {
-            return this.InvocationExpression(
-                this.IdentifierName(CSharp.SyntaxFacts.GetText(SyntaxKind.NameOfKeyword)),
-                expression);
+            // A bit hacky, but we need to actually run ParseToken on the "nameof" text as there's
+            // no other way to get a token back that has the appropriate internal bit set that indicates
+            // this has the .ContextualKind of SyntaxKind.NameOfKeyword.
+            var nameofToken = SyntaxFactory.ParseToken("nameof");
```

Comment:
❔ Can we cache the result of this to a static field?

Reply:
will do.